manufacturer-toolkit/src/main/java/org/sdo/sct/mt/OnDieCache.java

```diff
@@ -0,0 +1,239 @@
+      throws Exception {
+
+    if (sourceUrlList != null && !sourceUrlList.isEmpty()) {
+      String[] urls = sourceUrlList.split(",");
+      for (String url : urls) {
+        this.sourceUrl.add(new URL(url.trim()));
+      }
+    } else {
+      // defaults: the public facing sites containing OnDie artifacts
+      this.sourceUrl.add(new URL("https://tsci.intel.com/content/OnDieCA/certs/"));
+      this.sourceUrl.add(new URL("https://tsci.intel.com/content/OnDieCA/crls/"));
+    }
+
+    if (cacheDir != null) {
+      File cache = new File(cacheDir);
+      if (!cache.exists()) {
+        throw new IOException("OnDieCertCache: cache directory does not exist: " + cacheDir);
+      }
+      if (!cache.isDirectory()) {
+        throw new IOException("OnDieCertCache: cache directory must be a directory: " + cacheDir);
+      }
+
+      this.cacheDir = cacheDir;
+      this.autoUpdate = autoUpdate;
+      if (autoUpdate) {
+        // update local cache
+        copyFromUrlSources();
+      }
+      loadCacheMap();
+    }
+  }
+
+
+  /**
+   * Copy the certs and CRLs from the URL sources to the cache directory.
+   *
+   */
+  private void copyFromUrlSources() {
+
+    File cache = new File(cacheDir);
+    if (!cache.exists()) {
+      cache.mkdir();
+    }
+
+    try {
+      for (URL url : this.sourceUrl) {
+        URLConnection urlConn = url.openConnection();
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(
+            urlConn.getInputStream()))) {
+
+          // loop through all the href entries and for each .crl and .cer
+          // links, download the file and store locally
+          Document doc = Jsoup.connect(url.toString()).get();
+          Elements elements = doc.select("a[href]");
+          if (elements != null) {
+            for (Element e : elements) {
+              if (e != null) {
+                for (Attribute attr : e.attributes()) {
+                  if (attr.getKey().equals("href")) {
+                    String hrefValue = attr.getValue();
+                    if (hrefValue.contains(".cer") || hrefValue.contains(".crl")) {
+                      URL fileUrl = new URL(url, hrefValue);
+                      byte[] fileBytes = fileUrl.openConnection().getInputStream().readAllBytes();
+                      Files.write(Paths.get(cacheDir, hrefValue), fileBytes);
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    } catch (IOException ex) {
+      LoggerFactory.getLogger(getClass()).debug(ex.getMessage(), ex);
+    }
+  }
+
+  /**
+   * Loads the memory map of cache values from the cache directory.
+   *
+   * @throws Exception if error
+   */
+  private void loadCacheMap() throws IOException {
+    if (cacheDir != null) {
+      File cache = new File(cacheDir);
+
+      // Read each file and load into the hashmap
+      File[] files = new File(cache.getAbsolutePath()).listFiles();
+      if (files != null) {
```

Comment:
Following code layout might increase code readability.

if (files == null)
  return;

for (File file : files) {
  if (file.isDirectory())
     continue;
  if ...
}